blocks/api/parser.js

```diff
@@ -181,12 +181,14 @@ export function getAttributesFromDeprecatedVersion( blockType, innerHTML, attrib
 		try {
 			// Handle migration of older attributes into current version if necessary.
 			const deprecatedBlockAttributes = getBlockAttributes( deprecatedBlockType, innerHTML, attributes );
-			const migratedBlockAttributes = deprecatedBlockType.migrate ? deprecatedBlockType.migrate( deprecatedBlockAttributes ) : deprecatedBlockAttributes;
+			const migratedBlockAttributesAndInnerBlocks = deprecatedBlockType.migrate ?
+				deprecatedBlockType.migrate( { attributes: deprecatedBlockAttributes, innerBlocks } ) :
```

Comment:
Funny that to avoid breaking changes, we're introducing another breaking change here :)  (especially the return value)

Options here:

 - We're ok with this, assuming people didn't use `migrate` yet.
 - We're not ok and we try to support the old API. Here's how I would achieve it:
   - Instead of passing an object, pass the `innerBlocks` as a second argument.
   - If you want to update both attributes and innerBlocks, return an array with two elements, if you return just an object assume we update only the attributes.

Thoughs @aduth @mcsf

Reply:
I'm still forming an opinion, but leaning toward number 2. In practice, I doubt that `migrate` has been used a lot in the wild up till now, but that's entirely speculative (thus epistemically irresponsible? 😄), and I think it's bad karma to be changing a migration API at this stage.